test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -77,6 +79,70 @@ def date(error_code=None):
     return Any(isodate, error_code=error_code)
 
 
+_MODULE = re.compile(r"\bM\(([^)]+)\)")
+_LINK = re.compile(r"\bL\(([^)]+)\)")
+_URL = re.compile(r"\bU\(([^)]+)\)")
+_REF = re.compile(r"\bR\(([^)]+)\)")
+
+
+def _check_module_link(directive, content):
+    if not FULLY_QUALIFIED_COLLECTION_RESOURCE_RE.match(content):
+        raise _add_ansible_error_code(
+            Invalid('Directive "%s" must contain a FQCN' % directive), 'invalid-docs-markup')
+
+
+def _check_link(directive, content):
+    if ',' not in content:
+        raise _add_ansible_error_code(
+            Invalid('Directive "%s" must contain a comma' % directive), 'invalid-docs-markup')
+    idx = content.rindex(',')
+    title = content[:idx]
+    url = content[idx + 1:].lstrip(' ')
+    _check_url(directive, url)
+
+
+def _check_url(directive, content):
+    try:
+        parsed_url = urlparse(content)
+        if parsed_url.scheme not in ('', 'http', 'https'):
+            raise ValueError('Schema must be HTTP, HTTPS, or not specified')
+    except ValueError as exc:
+        raise _add_ansible_error_code(
+            Invalid('Directive "%s" must contain an URL' % directive), 'invalid-docs-markup')
+
+
+def _check_ref(directive, content):
+    if ',' not in content:
+        raise _add_ansible_error_code(
+            Invalid('Directive "%s" must contain a comma' % directive), 'invalid-docs-markup')
+
+
+def doc_string(v):
+    """Create a list of valid version for deprecation entries, current+4"""
```

Comment:
Looks like a copypasta - in `doc_string_or_strings` too.

Reply:
Indeed! Is now also fixed. Thanks!